apps/wallet-dashboard/components/dialogs/migration/MigrationDialog.tsx

```diff
@@ -32,18 +35,33 @@ export function MigrationDialog({
     isTimelocked,
 }: MigrationDialogProps): JSX.Element {
     const account = useCurrentAccount();
+    const [basicOutputs, setBasicOutputs] = useState<IotaObjectData[]>(basicOutputObjects);
+    const [nftOutputs, setNftOutputs] = useState<IotaObjectData[]>(nftOutputObjects);
+    const [isPartialMigration, setIsPartialMigration] = useState<boolean>(false);
     const [txDigest, setTxDigest] = useState<string | null>(null);
     const [view, setView] = useState<MigrationDialogView>(MigrationDialogView.Confirmation);
 
     const {
         data: migrateData,
         isPending: isMigrationPending,
         isError: isMigrationError,
-    } = useMigrationTransaction(account?.address || '', basicOutputObjects, nftOutputObjects);
-
+        error,
+    } = useMigrationTransaction(account?.address || '', basicOutputs, nftOutputs);
     const { mutateAsync: signAndExecuteTransaction, isPending: isSendingTransaction } =
         useSignAndExecuteTransaction();
 
+    useEffect(() => {
+        if (isMigrationError && error?.message.includes(MAX_SIZE_BYTES_ERROR)) {
+            const newBasicOutputs = basicOutputs.slice(0, -REDUCTION_STEP_SIZE);
+            const newNftOutputs = nftOutputs.slice(0, -REDUCTION_STEP_SIZE);
+
+            setBasicOutputs(newBasicOutputs);
+            setNftOutputs(newNftOutputs);
+            setIsPartialMigration(true);
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [isMigrationError]);
```

Comment:
```suggestion
const isPartialMigration = isMigrationError && error?.message.includes(MAX_SIZE_BYTES_ERROR);

    const [basicOutputs, nftOutputs] = useMemo(() => {
        if (isPartialMigration) {
            reductionSize.current += REDUCTION_STEP_SIZE;
            return [
                basicOutputObjects.slice(0, -reductionSize.current),
                nftOutputObjects.slice(0, -reductionSize.current)
            ]
        } else {
            return [[], []]
        }
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [isPartialMigration]);
```

Reply:
In the suggested changes the `IsPartialMigration` will change back to `false` once the possible transaction is calculated and there is no more errors, but this shouldn't be the case because once `isPartialMigration` is `true` than we use this flag to show the info message to the user.

Reply:
You can save `IsPartialMigration` in a `useRef` then, this way once its marked as `true` you will not mark it as `false` again